orangecontrib/spectroscopy/widgets/owpeakfit.py

```diff
@@ -349,28 +376,45 @@ def progress_interrupt(i: float):
         # happen when adding a preprocessor (there, commit() is called twice).
         # Wait 100 ms before processing - if a new task is started in meanwhile,
         # allow that is easily` cancelled.
-        for i in range(10):
-            time.sleep(0.005)
+        for _ in range(10):
+            time.sleep(0.010)
             progress_interrupt(0)
 
         model, parameters = create_composite_model(m_def)
 
         data_fits = data_anno = data_resid = None
         if data is not None and model is not None:
             orig_data = data
-            output = init_output_array(data, model, parameters)
+            output = []
             x = getx(data)
             n = len(data)
             fits = []
             residuals = []
-            for row in data:
-                i = row.row_index
-                out = model.fit(row.x, parameters, x=x)
-                add_result_to_output_array(output, i, out, x)
-                fits.append(out.eval(x=x))
-                residuals.append(out.residual)
-                progress_interrupt(i / n * 100)
-            data = fit_results_table(output, out, orig_data)
+
+            with multiprocessing.Pool(processes=None,
+                                      initializer=pool_initializer,
+                                      initargs=(model.dumps(), parameters, x)) as p:
+                res = p.map_async(pool_fit, data.X, chunksize=1)
+
+                def done():
+                    try:
+                        return n - res._number_left * res._chunksize
+                    except AttributeError:
+                        return 0
+
+                while not res.ready():
+                    progress_interrupt(done() / n * 99)
+                    res.wait(0.05)
+
+            fitsr = res.get()
+
+            for fit, bpar, fitted, resid in fitsr:
```

Comment:
What guarantees do we have here that the order of rows in the result matches the order of rows in the source table? Perhaps I am overly paranoid, but this is why I was using `row.id` in the previous code.

Reply:
The map function from the Pool guarantees the order.